tools/windows-node-installer/test/e2e/aws_test.go

```diff
@@ -1,90 +1,289 @@
-	awscredentials := os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
-	dir := os.Getenv("ARTIFACT_DIR")
+	kubeconfig     = os.Getenv("KUBECONFIG")
+	awscredentials = os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
+	dir            = os.Getenv("ARTIFACT_DIR")
 
-	// The e2e test uses Microsoft Windows Server 2019 Base with Containers image, m4.large type, and libra ssh key.
 	// The CI-operator uses AWS region `us-east-1` which has the corresponding image ID: ami-0b8d82dea356226d3 for
 	// Microsoft Windows Server 2019 Base with Containers.
+	imageID      = "ami-0b8d82dea356226d3"
+	instanceType = "m4.large"
+	sshKey       = "libra"
+
+	// awsProvider is setup as a variable for both creating, destroying, and tier down Windows instance.
+	awsProvider = SetupAWSCloudProvider()
+
+	// Set global variables for instance object, instance, security group,
+	// and infrastructure IDs so that once they are created,
+	// they will be used by all subsequent testing functions.
+	createdInstance   = &ec2.Instance{}
+	createdInstanceID = ""
+	createdSgID       = ""
+	infraID           = ""
+)
+
+// SetupAWSCloudProvider creates provider ofr Cloud interface and asserts type into AWS provider.
+// This is the first step of the e2e test and function exit with fatal error if failed.
+func SetupAWSCloudProvider() *awscp.AwsProvider {
+	// The e2e test uses Microsoft Windows Server 2019 Base with Containers image, m4.large type, and libra ssh key.
 	cloud, err := cloudprovider.CloudProviderFactory(kubeconfig, awscredentials, "default", dir,
-		"ami-0b8d82dea356226d3", "m4.large", "libra")
-	assert.NoError(t, err, "error creating clients")
+		imageID, instanceType, sshKey)
+	if err != nil {
+		log.Fatalf("unable to create an instance")
+	}
+
+	// Type assert to AWS so that we can test other functionality
+	awsProvider, ok := cloud.(*awscp.AwsProvider)
+	if !ok {
+		log.Fatalf("Error asserting cloudprovider to awsProvider")
+	}
+	return awsProvider
+}
 
-	err = cloud.CreateWindowsVM()
+// TestCreateWindowsInstance creates a Windows instance on AWS and assert no error from instance creation.
+// If it fails, abandon the rest of the test and clean up.
+func TestCreateWindowsInstance(t *testing.T) {
+	err := awsProvider.CreateWindowsVM()
 	assert.NoError(t, err, "error creating Windows instance")
 	if err != nil {
 		// TODO: Remove the lingering resources as part of test setup.
-		t.Fatalf("Error while creating a VM delete the virtual machine")
-	}
-	// Type Assert to AWS so that we can test other functionality
-	aws, ok := cloud.(*awscp.AwsProvider)
-	if !ok {
-		t.Fatal("Error asserting cloudprovider to aws")
+		assert.FailNow(t, "Error while creating a VM delete the virtual machine")
 	}
+}
+
+// TestInstallerJsonFileRead reads instance and security group IDs from windows-node-installer.json file and
+// updates the createdInstanceID and CreatedSgID global variables.
+// If it fails, abandon the rest of the test and clean up.
+func TestInstallerJsonFileRead(t *testing.T) {
```

Comment:
How are you guaranteeing ordering?

Reply:
You cannot rely on `go test` to run the test sequentially. Please explore [subtests](https://blog.golang.org/subtests) and come back to @ravisantoshgudimetla and me with your plan before you start working on implementing it.